dash/frontend/src/app/core/route-guards/auth.guard.ts

```diff
@@ -2,33 +2,58 @@ import { Injectable } from '@angular/core';
 import {CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree, Router} from '@angular/router';
 import {Observable, of} from 'rxjs';
 import {AlertService} from '@full-fledged/alerts';
-import {HttpClient} from '@angular/common/http';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
 import {catchError, map} from 'rxjs/operators';
 import {NgxUiLoaderService} from 'ngx-ui-loader';
+import { JwtAuthService } from '../services/jwt-auth.service';
+import { IServerResponse } from '../entities/IServerResponse';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthGuard implements CanActivate {
 
-  constructor(private readonly router: Router,
-              private alertService: AlertService,
-              private httpClient: HttpClient,
-              private loaderService: NgxUiLoaderService) {
-  }
+  constructor(
+    private readonly router: Router,
+    private alertService: AlertService,
+    private httpClient: HttpClient,
+    private loaderService: NgxUiLoaderService,
+    private jwtAuthService: JwtAuthService,
+  ) {}
 
   canActivate(
```

Comment:
reworked this method to use the /auth/check-status endpoint. it no longer relies on a 401 to tell whether the user is logged out.